pkg/operator/staticpod/controller/guard/guard_controller.go

```diff
@@ -289,13 +286,14 @@ func (c *GuardController) sync(ctx context.Context, syncCtx factory.SyncContext)
 			pod.Spec.NodeName = node.Name
 			pod.Spec.Containers[0].Image = c.installerPodImageFn()
 			pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Host = operands[node.Name].Status.PodIP
-			// The readyz port as string type is expected to be convertible into int!!!
-			readyzPort, err := strconv.Atoi(c.readyzPort)
-			if err != nil {
-				panic(err)
-			}
-			pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Port = intstr.FromInt(readyzPort)
-			pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Path = c.readyzEndpoint
+
+			pod.Spec.Containers[0].ReadinessProbe.HTTPGet.Port = intstr.FromString(c.readyZConf.Port)
```

Comment:
This change mean that we won't crash on config error anymore, but default to 0 and I don't think that's what we want since it hides the config error to the users.

Reply:
ah good point. There's `IntOrString.Parse(string)` , which does that exact same thing.